### ISSUE TITLE
Error when trying to list sources

### ISSUE THREAD

**Issue by ibagur** [cpacker/MemGPT#1587]
**Describe the bug**

After creating an external data source (a pdf document), the CLI command ` memgpt list sources` issues the error below (though the document can still be added later to the agent using `/attach` from within the agent session):

```
Traceback (most recent call last):
  File ".../.local/share/virtualenvs/MemGPT-npmQTUT2/bin/memgpt", line 8, in <module>
    sys.exit(app())
             ^^^^^
  File ".../.local/share/virtualenvs/MemGPT-npmQTUT2/lib/python3.11/site-packages/typer/main.py", line 328, in __call__
    raise e
  File ".../.local/share/virtualenvs/MemGPT-npmQTUT2/lib/python3.11/site-packages/typer/main.py", line 311, in __call__
    return get_command(self)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../.local/share/virtualenvs/MemGPT-npmQTUT2/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../.local/share/virtualenvs/MemGPT-npmQTUT2/lib/python3.11/site-packages/typer/core.py", line 783, in main
    return _main(
           ^^^^^^
  File ".../.local/share/virtualenvs/MemGPT-npmQTUT2/lib/python3.11/site-packages/typer/core.py", line 225, in _main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File ".../.local/share/virtualenvs/MemGPT-npmQTUT2/lib/python3.11/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../.local/share/virtualenvs/MemGPT-npmQTUT2/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../.local/share/virtualenvs/MemGPT-npmQTUT2/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../.local/share/virtualenvs/MemGPT-npmQTUT2/lib/python3.11/site-packages/typer/main.py", line 683, in wrapper
    return callback(**use_params)  # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^
  File ".../.../work/MemGPT/memgpt/cli/cli_config.py", line 1150, in list
    source.embedding_model,
    ^^^^^^^^^^^^^^^^^^^^^^
  File ".../.local/share/virtualenvs/MemGPT-npmQTUT2/lib/python3.11/site-packages/pydantic/main.py", line 828, in __getattr__
    raise AttributeError(f'{type(self).__name__!r} object has no attribute {item!r}')
AttributeError: 'SourceModel' object has no attribute 'embedding_model'. Did you mean: 'embedding_config'?
```

**Please describe your setup**
- [ ] How did you install memgpt?  `git clone` and then 'pip install -e .'
- [ ] Describe your setup
  - What's your OS (Windows/MacOS/Linux)? MacOS Sonoma 14.5
  - How are you running `memgpt`? Terminal CLI
  - MemGPT v0.3.21 (latest)
  - Virtual env running Python 3.11.4

**Screenshots**

**Additional context**
I am using openai backend with API key. I originally added a pdf source following instructions similar to the tutorial:
```
# we're saving the file as "memgpt_research_paper.pdf"
curl -L -o memgpt_research_paper.pdf https://arxiv.org/pdf/2310.08560.pdf

memgpt load directory --name memgpt_research_paper --input-files=memgpt_research_paper.pdf
```

**MemGPT Config**

```

[defaults]
preset = memgpt_chat
persona = sam_pov
human = basic

[model]
model = gpt-4
model_endpoint = https://api.openai.com/v1
model_endpoint_type = openai
context_window = 8192

[embedding]
embedding_endpoint_type = openai
embedding_endpoint = https://api.openai.com/v1
embedding_model = text-embedding-ada-002
embedding_dim = 1536
embedding_chunk_size = 300

[archival_storage]
type = chroma
path = /Users/inigo/.memgpt/chroma

[recall_storage]
type = sqlite
path = /Users/inigo/.memgpt

[metadata_storage]
type = sqlite
path = /Users/inigo/.memgpt

[version]
memgpt_version = 0.3.21

[client]
anon_clientid = 00000000-0000-0000-0000-000000000000


```

---

**Comment by sarahwooders** [comment]
Thanks for reporting this! We will try to fix ASAP.

**Comment by ibagur** [comment]
@sarahwooders many thanks! As an update to this issue, the 'sources' functionality seems to work fine over the deployed Docker web interface for MemGPT. In there I can add and later list the documents (pdf's in my case) added. They appear in the web interface and no error is raised. I guess then it might be a particular issue with the CLI implementation of the same functionality (maybe a missing parameter in a function call?). Thanks again!

**Comment by sarahwooders** [comment]
This should be fixed by #1596, but please re-open if you keep having the issue. 

The CLI and server implementations are often different (we are in the process of migrating everything to interact through a common client), which is probably why it was working on the web interface but not the CLI. The CLI was the first thing we had with MemGPT, so tends to have the oldest/messiest code 😅
